modules/nextflow/src/main/groovy/nextflow/processor/TaskConfig.groovy

```diff
@@ -345,8 +345,7 @@ class TaskConfig extends LazyMap implements Cloneable {
 
     int getMaxRetries() {
         def result = get('maxRetries')
-        def defResult = getErrorStrategy() == ErrorStrategy.RETRY ? 1 : 0
-        result ? result as int : defResult
+        result != null ? result as int : 1
```

Comment:
```suggestion
        result ? result as int : 1
```

Better to keep as it was before

Reply:
I changed because result can be an integer and 0 is considered as false. It was breaking the test when number of retries is set to 0.

Reply:
I think it has not too much sense but there was this possibility and it was in the test. Or we keep null check or we change the test.

Reply:
My reasoning here is to preserve the current behaviour that return `1` when max retries is set to `null` or `0`. The test was not taking into consideration the error strategy and therefore testing not the real usage `maxRetries` that's when `errorStrategy=='retry'`